src/lib/best_tip_prover/best_tip_prover.ml

```diff
@@ -26,10 +27,10 @@ end
 module Make (Inputs : Inputs_intf) :
   Coda_intf.Best_tip_prover_intf
   with type transition_frontier := Inputs.Transition_frontier.t
-   and type external_transition := Inputs.External_transition.t
+   and type external_transition := External_transition.t
    and type external_transition_with_initial_validation :=
-              Inputs.External_transition.with_initial_validation
-   and type verifier := Inputs.Verifier.t = struct
+              External_transition.with_initial_validation
+   and type verifier := Verifier.t = struct
```

Comment:
Now that these are all no longer part of the functor inputs, you can actually remove these abstract types entirely from the `Best_tip_prover_intf` signature. For example, for `external_transition`, you can remove the `type external_transition` from the signature and just `%s/external_transition/External_transition.t` (regexp) the rest of the signature. Then you wouldn't need to provide this type replacement everywhere we refer to the `Best_tip_prover_intf` signature.

Reply:
Oh I was meaning to do this and defunctoring other modules in a different PR (see pr description). Wasn't sure how big of a change that'd be. But I guess I could do just this one now and the defunctoring other modules after persistence lands